lib/ansible/plugins/inventory/onesixtyone.py

```diff
@@ -0,0 +1,251 @@
+            if path.endswith(maybe):
+                return True
+
+    def parse(self, inventory, loader, path, cache=False):
+        if not HAS_NETADDR:
+            raise AnsibleParserError('this inventory plugin requires the netaddr library. Try: pip install netaddr')
+
+        if not HAS_GEVENT:
+            raise AnsibleParserError('this inventory plugin requires the gevent library. Try: pip install gevent')
+
+        # Initialize Superclass Method
+        super(InventoryModule, self).parse(inventory, loader, path, cache=cache)
+
+        # Find the onesixtyone binary
+        try:
+            self._onesixtyone = get_bin_path(self.NAME, True)
+        except Exception:
+            raise AnsibleParserError('onesixtyone inventory plugin requires the onesixtyone cli tool to work.')
+
+        # Read Configuration
+        self._read_config_data(path)
+
+        try:
+            # Using gevent to parallelize reverse dns lookups and adding entries to inventory
+            jobs = [gevent.spawn(self._add_device, self, device) for device in self._discovery()]
+            gevent.joinall(jobs, timeout=20)
+
+        # Exception raised by gevent
+        except LoopExit as e:
+            raise AnsibleParserError('gevent library exception, this should not happen: %s' % to_native(e))
+
+    def _match_platform(self, text):
+        """ Matches platforms against text in sysDescr and return the platform.
+
+        Keyword arguments:
+        text -- sysdescr text
+        """
+        platforms = {
+            'asa': re.compile(r'Cisco\sAdaptive\sSecurity\sAppliance'),
+            'ios': re.compile(r'Cisco\sIOS\sSoftware'),
+            'iosxr': re.compile(r'Cisco\sIOS\sXR\sSoftware'),
+            'nxos': re.compile(r'Cisco\sNexus\sOperating\sSystem'),
+            'dellos': re.compile(r'Dell\sApplication\sSoftware'),
+            'junos': re.compile(r'JUNOS'),
+            'aruba': re.compile(r'Aruba\sOperating\sSystem\sSoftware'),
+            'eos': re.compile(r'Arista'),
+        }
+        platforms.update(self.get_option('platform'))
+
+        for platform, rule in platforms.items():
+            if rule.match(text):
+                return platform
+
+        return 'unknown'
+
+    def _discovery(self):
+        """ Run onesixtyone and parse the output to extract host, community and sysdescr """
+        # setup command
+        cmd = [self._onesixtyone]
+
+        #                                ip.ip.ip                       [community] <text>
+        find_entry = re.compile(r'^(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\s\[(\w+)\]\s(.*)')
+
+        # Create communities temporary file
+        with tempfile.NamedTemporaryFile() as communities_tempfile:
+            communities = self._options.get("communities")
+            [communities_tempfile.write("%s\n" % community) for community in communities]
+            communities_tempfile.flush()
+
+            # add command-line option -c
+            cmd.append('-c')
```

Comment:
It'd be probably better readable with
```suggestion
            cmd.extend(('-c', communities_tempfile.name))
```
instead